### DISCUSSION TITLE
Pixel XL AR mode black screen iFrame

### DISCUSSION THREAD

**Discussion by dmq-ar** [google/model-viewer#1907] [Q&A]
### Description

My investigation of #1318 

My developer page not showing AR mode, but the same code works great on [our customer's website](https://dantonehome.ru/catalog/new/divan-narvik/).

![image](https://user-images.githubusercontent.com/62610948/89135415-1544c480-d536-11ea-8d29-78474c3a5abc.png)

The idea is very simple: add a link to JS code located on our server and just put an iframe in the web page. Something like this. 
```
<!doctype html>
<html>
​
<head>
    <script src="https://code.jquery.com/jquery-3.4.1.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
    <script src="https://rosatom-bdcd9.firebaseapp.com/ModelViewer.js"></script>
    <link href="https://rosatom-bdcd9.firebaseapp.com/style.css" rel="stylesheet">
</head>
​
<body>
    <iframe id="modelViewerIframe" src="" scrolling="no" frameborder="0"></iframe>
</body>
​
<script>
    loadModel("Atom"); //ID here
</script>
```

Also, if I change [<iframe>](https://rosatom-bdcd9.firebaseapp.com/#Atom) to [model-viewer](https://rosatom-bdcd9.firebaseapp.com/test.html) on my website AR mode works fine. But the same <iframe> block on Dantone's website also works fine (interesting fact - it opens AR mode not the same way like model-viewer always does, it looks like the thing with AR animals in Google search). 

![image](https://user-images.githubusercontent.com/62610948/89135470-8be1c200-d536-11ea-8403-33224da62408.png)

So, what's going on? o_O 
Unfortunately, I'm not a developer of that website, I've just created the HTML snippet above. I tried to inspect Dantone's webpage, but there are so many things in the code... As not being a web developer I feel confused.

P.S. iOS works as expected. 

### Version
```
<script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
<script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
```

### Browser Affected

- [x] Chrome

### OS

- [x] Android

**Comment by elalish** [comment]
As best I can tell, what you've found is that the model works in 3D, SceneViewer (this is the AR Animals app) and QuickLook (the iOS app), but not WebXR in an iFrame. As you found, that is a known issue and one we intend to fix. If you need to use an iFrame, for now you can specify `ar-modes="scene-viewer quick-look"` to skip the webXR bug.

**Comment by dmq-ar** [comment]
> If you need to use an iFrame, for now you can specify ar-modes="scene-viewer quick-look" to skip the webXR bug.

Thanks, it did a trick! However, the 3D model is too large for some reason... not the same size as in iOS quick-look. I guess it's a "feature" of USDZ converter. 

> As best I can tell, what you've found is that the model works in 3D, SceneViewer (this is the AR Animals app) and QuickLook (the iOS app), but not WebXR in an iFrame.

The main idea I'd like to share that for some reason the same code forces SceneViewer or WebXR mode on different websites without a fix above. Why?

**Comment by elalish** [comment]
Check that your address is https; http won't allow webXR to work, and it'll fall back to sceneViewer. This is a common problem as Android likes to default to http.

**Comment by dmq-ar** [comment]
I'm using https only for hosting and file server
